backend/leanstore/storage/buffer-manager/BufferFrame.hpp

```diff
@@ -81,7 +81,10 @@ struct BufferFrame {
    // -------------------------------------------------------------------------------------
    bool operator==(const BufferFrame& other) { return this == &other; }
    // -------------------------------------------------------------------------------------
-   inline bool isDirty() const { return page.PLSN != header.last_written_plsn; }
+   inline bool isDirty() const { return 
+      page.PLSN != 0 && // marked as dirty
```

Comment:
PLSN will increase monotonically and will be increased here:
https://github.com/leanstore/leanstore/blob/master/backend/leanstore/sync-primitives/PageGuard.hpp#L122C1-L127C23
That means, your first condition is inclusive with `page.PLSN != header.last_written_plsn`

Reply:
If the page is only read but not written, it should not be flushed. My experiments also see write volume for read-only workload.